src/include/elektra/error.h

```diff
@@ -22,10 +22,17 @@ typedef struct _ElektraError ElektraError;
 
 typedef void (*ElektraErrorHandler) (ElektraError * error);
 
+ElektraError * elektraErrorPureWarning (void);
```

Comment:
What is PureWarning? Design decision for API missing.

Reply:
This change was actually done by @kodebach to make the highlevel conform with the new error code system. CLion tells me it is unused though.

Reply:
https://github.com/ElektraInitiative/libelektra/blob/cbd85e11236b1d0c0eadb77d3cd0f2a8709470be/src/libs/highlevel/elektra_error.c#L208-L214

The doxygen documentation of `elektraErrorPureWarning` describes exactly what this is used for. We need it because the highlevel API only has the `ElektraError` struct and no separate `ElektraWarning` struct.

> CLion tells me it is unused though.

That is a mistake on my part, it should be used here (instead of the `elektraErrorCreate` call):

https://github.com/ElektraInitiative/libelektra/blob/cbd85e11236b1d0c0eadb77d3cd0f2a8709470be/src/libs/highlevel/elektra_error.c#L98-L101

Reply:
Fixed in https://github.com/ElektraInitiative/libelektra/pull/2686/commits/4966b8a830880bd39bffd6f036936933976f04e3

Reply:
Fixed in https://github.com/ElektraInitiative/libelektra/pull/2686/commits/9df13eb2affcc573879c005f066939abf1a81210

Reply:
Why is this change in the PR of @Piankero then?

Reply:
Because the changes to low-level errors required that the high-level errors API is changed as well.